app/src/main/java/com/example/poste/activities/EditPostActivity.java

```diff
@@ -2,52 +2,32 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
-import android.app.AlertDialog;
-import android.app.ProgressDialog;
-import android.content.DialogInterface;
 import android.content.Intent;
-import android.os.AsyncTask;
 import android.os.Bundle;
 import android.util.Log;
-import android.view.LayoutInflater;
-import android.view.MenuItem;
-import android.view.View;
-import android.view.ViewGroup;
-import android.view.Window;
-import android.view.WindowManager;
-import android.widget.AdapterView;
-import android.widget.ArrayAdapter;
-import android.widget.BaseAdapter;
 import android.widget.Button;
 import android.widget.EditText;
-import android.widget.ImageView;
-import android.widget.PopupMenu;
-import android.widget.Spinner;
-import android.widget.TextView;
 import android.widget.Toast;
 
-import com.example.poste.PosteApplication;
 import com.example.poste.R;
-import com.example.poste.adapters.FolderAdapter;
-import com.example.poste.api.poste.API;
-import com.example.poste.api.poste.exceptions.APIException;
-import com.example.poste.api.poste.models.Folder;
-import com.example.poste.api.poste.models.FolderAccess;
-import com.example.poste.api.poste.models.User;
-import com.example.poste.api.poste.exceptions.IncompleteRequestException;
-import com.example.poste.api.poste.exceptions.MalformedResponseException;
-import com.example.poste.api.poste.models.Folder;
-import com.example.poste.api.poste.models.FolderAccess;
-import com.example.poste.api.poste.models.Post;
-
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.stream.Collectors;
+import com.example.poste.http.EditPostRequest;
+import com.example.poste.http.LoginRequest;
+import com.example.poste.http.MyApiService;
+import com.example.poste.http.RetrofitClient;
+import com.example.poste.models.Folder;
+import com.example.poste.models.Post;
+import com.example.poste.models.User;
+import com.example.poste.utils.DebugUtils;
+
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 /**
  * The EditPostActivity class adds functionality to the activity_edit_post.xml layout
+ *
+ * The activity that calls this activity, do a .putExtra to pass the postId
```

Comment:
Can you add an example here, showing how to call the activity with the putExtra? Do we need to send just postId, or also folderId?

Reply:
I have included the example.

Current implementation needs both folderId and postId.
I agree it is possible to do it with only postId... For some reason it just didn't occur to me when I was coding this